gax/src/main/java/com/google/api/gax/rpc/ApiClientHeaderProvider.java

```diff
@@ -92,44 +123,45 @@ public String getJvmToken() {
       return jvmToken;
     }
 
-    public Builder setJvmToken(String jvmToken) {
-      this.jvmToken = jvmToken;
+    public Builder setJvmToken(String name, String version) {
+      this.jvmToken = constructToken(name, "gl-java", version);
       return this;
     }
 
     public String getClientLibToken() {
       return clientLibToken;
     }
 
-    public Builder setClientLibToken(String clientLibToken) {
-      this.clientLibToken = clientLibToken;
+    public Builder setClientLibToken(String name, String version) {
+      this.clientLibToken = constructToken(name, "gl-java", version);
+      ;
       return this;
     }
 
     public String getGeneratedLibToken() {
       return generatedLibToken;
     }
 
-    public Builder setGeneratedLibToken(String generatedLibToken) {
-      this.generatedLibToken = generatedLibToken;
+    public Builder setGeneratedLibToken(String name, String version) {
```

Comment:
It feels weird for the key to be defaultable

Reply:
Why?

Reply:
1) Because
2) It's the first parameter, generally defaulting is done with parameters near the end instead of the beginning

Reply:
removed token name parameter for all token except generatedLibToken (in case if non-toolkit it used), clientLibToken (in case if client is not from google-cloud-java), transportToken (already different: grpc or httpjson).